File: samples/client/petstore/java/src/main/java/org/openapijsonschematools/client/schemas/validation/LengthValidator.java
package org.openapijsonschematools.client.schemas.validation;

import java.text.BreakIterator;

public abstract class LengthValidator {
    protected int getLength(String text) {
        int graphemeCount = 0;
        BreakIterator graphemeCounter = BreakIterator
                .getCharacterInstance();
        graphemeCounter.setText(text);
        while (graphemeCounter.next() != BreakIterator.DONE) {
            graphemeCount++;
        }
        return graphemeCount;
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/LengthValidator.kt
package org.openapijsonschematools.client.schemas.validation

import java.text.BreakIterator

abstract class LengthValidator {
    protected fun getLength(text: String?): Int {
        var graphemeCount = 0
        val graphemeCounter = BreakIterator
            .getCharacterInstance()
        graphemeCounter.setText(text)
        while (graphemeCounter.next() != BreakIterator.DONE) {
            graphemeCount++
        }
        return graphemeCount
    }
}
